### ISSUE TITLE
False positive E0013 (Static and const variables can refer to other const variables.)

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9264]
Originally posted at https://youtrack.jetbrains.com/issue/CPP-30298/rust-plugin-Invalid-Const-x-cannot-refer-to-static-y-E0013

## Environment

* **IntelliJ Rust plugin version:** 0.4.178.4836-222-nightly
* **Rust toolchain version:** 1.65.0-nightly (015a824f2 2022-08-22) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2.1 (CL-222.3739.54)
* **Operating system:** macOS 12.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
<img width="587" alt="image" src="https://user-images.githubusercontent.com/49211026/187212490-cbe45845-ff21-4dea-8cb2-d565883b0e0f.png">

## Steps to reproduce
```rust
use std::sync::atomic::{AtomicU32, Ordering};

pub struct UnitTest {
    pub f: fn(),
}

const TEST1: UnitTest = UnitTest {
    f: || {
        static COUNTER: AtomicU32 = AtomicU32::new(0);
        COUNTER.store(0, Ordering::Relaxed);
    }
};

fn main() {
    (TEST1.f)();
    println!("Hello, world!");
}

```
